src/main/java/eu/arrowhead/blacklist/jpa/service/EntryDbService.java

```diff
@@ -0,0 +1,227 @@
+import eu.arrowhead.blacklist.jpa.repository.EntryRepository;
+import eu.arrowhead.blacklist.service.dto.BlacklistCreateRequestDTO;
+import eu.arrowhead.blacklist.service.dto.enums.Mode;
+import eu.arrowhead.common.Utilities;
+import eu.arrowhead.common.exception.InternalServerError;
+
+@Service
+public class EntryDbService {
+
+	//=================================================================================================
+	// members
+
+	@Autowired
+	private EntryRepository entryRepo;
+
+	private final Logger logger = LogManager.getLogger(this.getClass());
+
+	private static final Object LOCK = new Object();
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	public List<Entry> createBulk(final List<BlacklistCreateRequestDTO> candidates, final String requesterName) {
+		logger.debug("createBulk started...");
+
+		try {
+			return entryRepo.saveAllAndFlush(createEntriesFromDTOs(candidates, requesterName));
+		} catch (final Exception ex) {
+			logger.error(ex.getMessage());
+			logger.debug(ex);
+			throw new InternalServerError("Database operation error");
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	public Page<Entry> getPageByFilters(final PageRequest pagination, final List<String> systemNames, final Mode mode,
+			final List<String> issuers, final List<String> revokers, final String reason, final ZonedDateTime alivesAt) {
+		logger.debug("getByFilters started...");
+		Assert.notNull(pagination, "page is null");
+
+		try {
+			Page<Entry> entries = null;
+
+			// Without filters
+			if (Utilities.isEmpty(systemNames)
+					&& mode == null
+					&& Utilities.isEmpty(issuers)
+					&& Utilities.isEmpty(revokers)
+					&& Utilities.isEmpty(reason)
+					&& alivesAt == null) {
+				entries = entryRepo.findAll(pagination);
+			}
+
+			// With filters
+			if (entries == null) {
+				synchronized (LOCK) {
+					final List<Long> matchingIDs = new ArrayList<>();
+					final List<Entry> toFilter = Utilities.isEmpty(systemNames) ? entryRepo.findAll() : entryRepo.findAllBySystemNameIn(systemNames);
+					for (final Entry entry : toFilter) {
+						if (mode != null && !modeMatch(mode, entry.getActive())) {
+							continue;
+						}
+						if (!Utilities.isEmpty(issuers) && !issuers.contains(entry.getCreatedBy())) {
+							continue;
+						}
+						if (!Utilities.isEmpty(revokers) && (entry.getRevokedBy() == null || !revokers.contains(entry.getRevokedBy()))) {
+							continue;
+						}
+						// reason match is case insensitive
+						if (!Utilities.isEmpty(reason) && !entry.getReason().toLowerCase().contains(reason.toLowerCase())) {
+							continue;
+						}
+						if (entry.getExpiresAt() != null && alivesAt != null && !(alivesAt.isBefore(entry.getExpiresAt()) || alivesAt.isEqual(entry.getExpiresAt()))) {
```

Comment:
this condition is not good. If an entry is inactive, it is not alive even if is expired at the specified moment.